### ISSUE TITLE
fix: `mason_logger` `prompt` with default does not accept default on Windows

### ISSUE THREAD

**Issue by bryanoltman** [felangel/mason#1189]
**Description**

When using `mason_logger`'s `prompt` function with a default value, pressing the enter key without entering any text should cause the `prompt` function to return the default value. However, on windows in powershell, Logger seems to not register the return value.

**Steps To Reproduce**

Create a new Dart app with `dart create`. Add `mason_logger` as a dependency and replace `bin\project_name.dart` with:

```dart
import 'package:mason_logger/mason_logger.dart';

void main(List<String> arguments) {
  final logger = Logger();
  final name = logger.prompt('What is your name?', defaultValue: 'John Doe');
  print('Hello, $name!');
}
```

On Windows, in Powershell, run `dart run ./project_name.dart` and press "Enter" when prompted. Note that this does not proceed with the default value.

**Notes**

- This works in the powershell embedded in VS Code, but not in a dedicated powershell window.
- If I press Enter multiple times and then enter non-empty text, mason will use that text. Ex:

```
$ dart run .\mason_logger_repro.dart my_command
What is your name? (John Doe)


What is your name? (John Doe) d
Hello, d!
```

**Comment by codesculpture** [comment]
Seems this issue is from `readLineSync` from `dart:io`.

**Comment by mraleph** [comment]
This is a bug in [Windows Terminal](https://github.com/microsoft/terminal/issues/16223). You should probably close it. I don't think we want to work around this on the Dart side.

**Comment by codesculpture** [comment]
One small question i had in my mind for long, as above issue is actually from windows but it still affects the users of the mason_logger, such issues can be closed here since if one user might face this issue and will probably look for does it already been reported and will fail to see theres such issue existed since its actually closed here in mason_logger. Is it normal or atleast okay to do?

**Comment by codesculpture** [comment]
Hmm, but that doesnt work if they only search on open issues, i dont know whether people will look on "issues" rather than "open issues"

**Comment by codesculpture** [comment]
This has been fixed in windows terminal v1.18.2 (tested) 🎉
